crates/subspace-core-primitives/src/pieces.rs

```diff
@@ -143,7 +143,7 @@ impl PieceIndex {
     /// Returns the next source piece index
     #[inline]
     pub const fn next_source_index(&self) -> PieceIndex {
-        PieceIndex(self.0.next_multiple_of(Self::source_ratio()))
+        PieceIndex(self.0 + Self::source_ratio() - (self.0 % Self::source_ratio()))
```

Comment:
I do not understand why subtraction is needed here, it'll always be 0.

Also the whole `source_ratio` doesn't really make sense because it only works for cases when numerator is 1. If we had 2/3 it would return 1, which would be wrong, so if this code was to be fixed, it'd be fixed to account for that.

Reply:
The subtraction is needed to calculate the next source piece after a parity piece.

Removing `source_ratio()` is a larger change which isn't needed to fix this bug, I'll push it in another PR.

Reply:
`source_ratio()` is 2, source indices are always even numbers, hence you're always subtracting 0. How would you describe what this code is doing with regular English?

Reply:
If you start with a source piece index, add 2 to go to the next source piece index.
If you start with a parity piece index, add 2 to go to the next source/parity piece pair, then subtract 1 to go to the source piece index in that pair.